lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -1228,164 +1237,201 @@ namespace Js
+        }
+        else
+        {
+            if (GetUserDefaultLocaleName(defaultLocale, _countof(defaultLocale)) != 0)
             {
-                aLeft = const_cast<char16*>(str1->GetSz());
-                size1 = str1->GetLength();
+                locale = defaultLocale;
             }
-            if (aRight == nullptr)
+            else
             {
-                aRight = const_cast<char16*>(str2->GetSz());
-                size2 = str2->GetLength();
+    #if defined(INTL_WINGLOB)
+                // win32 GetUserDefaultLocaleName returns its error through GetLastError
+                JavascriptError::MapAndThrowError(scriptContext, HRESULT_FROM_WIN32(GetLastError()));
+    #elif defined(INTL_ICU)
+                // TODO(jahorto): what error should we throw here for INTL_ICU?
+                JavascriptError::MapAndThrowError(scriptContext, E_FAIL);
+    #endif
             }
+        }
 
-#ifdef _WIN32
-            // xplat-todo: Need to replace this with platform-agnostic API
-            compareResult = CompareStringEx(givenLocale != nullptr ? givenLocale : defaultLocale, compareFlags, aLeft, size1, aRight, size2, NULL, NULL, 0);
-#else
-            // FIXME (doilij): when CompareStringEx is implemented in PlatformAgnostic, call that function here
-            compareResult = 2; // 2 means strings are equal (reasonable default)
-#endif
+        Assert(locale != nullptr);
+        Assert((int)sensitivity >= 0 && sensitivity < CollatorSensitivity::Max);
+        Assert((int)caseFirst >= 0 && caseFirst < CollatorCaseFirst::Max);
 
-            // Get the last error code so that it won't be affected by END_TEMP_ALLOCATOR.
-            if (compareResult == 0)
-            {
-                lastError = GetLastError();
-            }
+        BEGIN_TEMP_ALLOCATOR(tempAllocator, scriptContext, _u("EntryIntl_CompareString"));
+
+        const char16 *normalized1 = nullptr;
+        charcount_t normalized1Length = 0;
+        if (UnicodeText::IsNormalizedString(UnicodeText::NormalizationForm::C, str1->GetSz(), str1->GetLength()))
+        {
+            normalized1 = str1->GetSz();
+            normalized1Length = str1->GetLength();
         }
+        else
+        {
+            normalized1 = str1->GetNormalizedString(UnicodeText::NormalizationForm::C, tempAllocator, normalized1Length);
+        }
+
+        const char16 *normalized2 = nullptr;
+        charcount_t normalized2Length = 0;
+        if (UnicodeText::IsNormalizedString(UnicodeText::NormalizationForm::C, str2->GetSz(), str2->GetLength()))
+        {
+            normalized2 = str2->GetSz();
+            normalized2Length = str2->GetLength();
+        }
+        else
+        {
+            normalized2 = str2->GetNormalizedString(UnicodeText::NormalizationForm::C, tempAllocator, normalized2Length);
+        }
+
+        // CompareStringEx on Windows returns 0 for error, 1 if less, 2 if equal, 3 if greater
+        // Default to the strings being equal
```

Comment:
>// Default to the strings being equal [](start = 8, length = 37)

Should probably add the comment about ordering of == versus  a < b being true for all a, b. (Explaining why == is the default value.)

i.e. sorting with = causes no change in the order but converges, whereas < would cause an infinite loop. See my previous comment replying to Taylor here.